tincr/io/library/genCellLibrary.tcl

```diff
@@ -291,6 +291,110 @@ proc processMacroCell {c s fo} {
     }
 }
 
+proc doPorts { fo } {
+
+    puts "Doing ports..."
+
+    set lc [get_lib_cells]
+    set dict [uniqueSites]
+
+    puts $fo "    <cell>"
+    puts $fo "      <type>IPORT</type>"
+    puts $fo "          <is_port/>"
+    puts $fo "      <level>LEAF</level>"
+    puts $fo "      <pins>"
+    puts $fo "        <pin>"
+    puts $fo "          <name>PAD</name>"
+    puts $fo "          <direction>output</direction>"
+    puts $fo "        </pin>"
+    puts $fo "      </pins>"
+    puts $fo "      <bels>"
+    
+    dict for {type site} $dict {
```

Comment:
In these loops, I would add a check like: 
```
if {![get_property IS_PAD $site]} {
    continue;
}
```

To skip over sites that are not pads, and so we don't have to waste time searching bels that aren't no pad sites. The same applies for the loops below them as well.